### PR TITLE
=doc #15677 Document how to use native-packager

### PR DESCRIPTION
**Pull request by patriknw**
- replace docs of AkkaAppPackaging with JavaServerAppPackaging,
  since the former doesn't have the same flexibility and quality
- warn about microkernel deprecation

### CODE REVIEW

**Comment by bantonsson** [comment]
@patriknw I think you should `s/latter/former/` in the PR comment since it's the `AkkaAppPackaging` that is not so good.

LGTM

**Comment by patriknw** [comment]
@bantonsson good catch :)

**Comment by ktoso** [comment]
This would be a nice candidate to plugin ConductR's "all in one" plugin: https://github.com/sbt/sbt-typesafe-conductr but later, once it's fully completed. New issue?

File: akka-docs/rst/scala/microkernel.rst [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
 The purpose of the Akka Microkernel is to offer a bundling mechanism so that you can distribute
 an Akka application as a single payload, without the need to run in a Java Application Server or manually
 having to create a launcher script.
 
-The Akka Microkernel is included in the Akka download found at `downloads`_.
-
-.. _downloads: http://akka.io/downloads
+.. warning:: Akka Microkernel will be deprecated and removed. It will be replaced by using an ordinary
+   user defined main class and packaging with `sbt-native-packager <https://github.com/sbt/sbt-native-packager>`_
+   or `Typesafe ConductR <http://typesafe.com/products/conductr>`_.
```
Comment (ktoso) [inline akka-docs/rst/scala/microkernel.rst]: :+1:

**Comment by drewhk** [comment]
LGTM. We might want to extend the docs on packaging in the future and provide a fully fledged native-packager template, too.

File: akka-samples/akka-sample-hello-kernel/build.sbt [hunk 2]
```diff
@@ -1,12 +1,28 @@
-import NativePackagerKeys._
```
Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: Nice!

File: akka-samples/akka-sample-hello-kernel/build.sbt [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+  "ch.qos.logback" % "logback-classic" % "1.0.7"
 )
+
+mainClass in Compile := Some("akka.kernel.Main")
+
+enablePlugins(JavaServerAppPackaging)
+
+mappings in Universal ++= {
+  // copy additional scripts from bin directory
+  directory("bin") ++
```
Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: That shouldn't be necessary.

File: akka-samples/akka-sample-hello-kernel/build.sbt [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
+  // copy configuration files
+  contentOf("src/main/resources").toMap.mapValues("config/" + _)
+}
+
+// add 'config' directory first in the classpath of the start script
```
Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: This is interesting. I would rather recommended setting the config file locations via CLI parameters. This won't play nice with the `LauncherPackaging` or `ClasspathJar`

Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: That's not true. It should work, but may look awkward

Comment (patriknw) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: Note that this is instructions/sample of how to build something that is close to what the Akka dist plugin did for Akka Microkernel. I agree that this packaging is not the best thing we can do, but that is for another pull request targeted to master when Akka Microkernel is deprecated for removal.

File: akka-samples/akka-sample-hello-kernel/build.sbt (hunk 3 above)
Comment (patriknw) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: not necessary, but an example of how you can copy things if needed.
Not uncommon to have additional scripts, I think.

Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: Yep, but the `src/universal/bin` directory is always included. So this example maybe a bit misleading. We had this problem in our docs as well, when describing the app archetype as people just copy and pasted everything and created a mess :(

Maybe change it to `data-formats` or something else. Sorry for stressing this line so much, but in my experience users tend to use the hard way if they spot it :p

File: akka-samples/akka-sample-hello-kernel/build.sbt (hunk 4 above)
Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: Good point. Transition should be as painless as possible.

File: akka-samples/akka-sample-hello-kernel/build.sbt (hunk 3 above)
Comment (patriknw) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: ok, I'll change, thanks

File: akka-samples/akka-sample-hello-kernel/build.sbt (hunk 4 above)
Comment (patriknw) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: thanks for reviewing

Comment (muuki88) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: you are welcome :) Good docs mean less issues on sbt-native-packager ;)

**Comment by hepin1989** [comment]
it would be better to add something about mapping in the packaging and how to add external folder to the classpath.

File: akka-samples/akka-sample-hello-kernel/build.sbt (hunk 4 above)
Comment (patriknw) [inline akka-samples/akka-sample-hello-kernel/build.sbt]: I would like to hear your opinion about https://github.com/sbt/sbt-native-packager/issues/521. Let's discuss there.

**Pull request merged**